File: CodingTest/add_int/anotherCode.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(int a, int b);

int main() {
    printf("%lld\n", solution(6, 3));
    printf("%lld\n", solution(4, 9));
    printf("%lld\n", solution(5, 5));

    return 0;
}

long long solution(int a, int b) {
    long long answer = 0;
    
    //a는 무조건 작은 수, b는 무조건 큰수로 바꿔버리기
    if (a > b) {
        int temp = b;
        b = a;
        a = temp;
    }
    for (int i = a; i <= b; i++) {
        answer += i;
    }
    return answer;
}

File: CodingTest/add_int/anotherCode.go
package main

import "fmt"

func main() {
	fmt.Println(solution(6, 3))
	fmt.Println(solution(4, 9))
	fmt.Println(solution(5, 5))
}

// 정수 합 공식 이용하기
func solution(a int, b int) int64 {
	A := int64(a)
	B := int64(b)
	if A > B {
		return (A + B) * (A - B + 1) / 2
	} else if A < B {
		return (B + A) * (B - A + 1) / 2
	} else {
		return A
	}
}
